### PR TITLE
Add CLI, transpile vectors

### PR DESCRIPTION
**Pull request by teodorlu**
This PR proposes a CLI for `clj2el`. Example usage:

```
$ ls
source.clj  source2.clj
$ cat source.clj
(defn foo [x & xs] xs)

(inc 2)

(map inc [1 2 3])
$ cat source.clj | clj2el
(defun foo (x &rest xs) xs)

(1+ 2)

(mapcar #'1+ (vector 1 2 3))
$ cat source2.clj
[1 2 3]
$ cat source2.clj | clj2el
(vector 1 2 3)

```

### CODE REVIEW

File: src/clj2el/cli.clj [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  (str/join "\n" (map prn-str forms)))
+
+(defn pprint-str [form] (with-out-str (pprint form)))
+
+(defn pprint-str-multiple-forms [forms]
+  (str/join "\n" (map pprint-str forms)))
+
+(defn -main [& args]
+  (let [argset (set args)
+        multiple? (contains? argset "--multiple") ;; assuming we don't want to depend on babashka.cli
```
Comment (borkdude) [inline src/clj2el/cli.clj]: You can change the main-opts in the `bbin/bin` arguments to `-x cljel.exec` and then make a function that recieves a map

File: src/clj2el/cli.clj [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        pretty? (contains? argset "--pretty")
+        read-fn (if multiple? read-multiple-forms edn/read-string)
```
Comment (borkdude) [inline src/clj2el/cli.clj]: I think it's better to read like it's done in the web-app with edamame: parse-string-all

Comment (teodorlu) [reply src/clj2el/cli.clj]: Agreed, manually adding [] to the string to parse didn't feel right.

Comment (borkdude) [reply src/clj2el/cli.clj]: https://github.com/borkdude/clj2el/blob/b760eb9dae7d19ae294bef1fe4ff0179cc903818/web/src/clj2el/web.cljs#L17

Comment (borkdude) [reply src/clj2el/cli.clj]: We could also add an api function `clj2el-str` which takes a string and produces a string and then expose this edamame parsing over there instead

File: src/clj2el/cli.clj (hunk 1 above)
Comment (teodorlu) [reply src/clj2el/cli.clj]: Huh, good point. I didn't think of that. Thanks!

Though seeing what you did with the web interface makes me question whether we really need those options. I think `(edamame.core/parse-string-all s {:all true})` is a good default, and I can't really defend the need more configuration.

File: src/clj2el/cli.clj (hunk 2 above)
Comment (teodorlu) [reply src/clj2el/cli.clj]: Using edamame now. Everything just got easier!

File: src/clj2el/cli.clj [hunk 3]
```diff
@@ -12,5 +12,4 @@
   (str/join "\n" (map #(with-out-str (pprint %)) forms)))
 
 (defn -main [& _args]
-  (let [clj2el-fn (partial map clj2el)]
-    (-> *in* slurp read-str-multiple clj2el-fn write-str-multiple print)))
+  (->> *in* slurp read-str-multiple (map clj2el) write-str-multiple print))
```
Comment (borkdude) [inline src/clj2el/cli.clj]: for JVM Clojure we also need to call flush at the end

File: src/clj2el/cli.clj (hunk 1 above)
Comment (teodorlu) [reply src/clj2el/cli.clj]: I swapped out `-m` for `-x` anyway, I don't see any arguments for `-m` when we're running on babashka, and babashka.cli is embedded.

File: src/clj2el/cli.clj (hunk 3 above)
Comment (teodorlu) [reply src/clj2el/cli.clj]: just like this? I assume calling `(flush)` doesn't break anything when we're running on babashka?

```clj
(defn exec [_opts]
  (->> *in* slurp read-str-multiple (map clj2el) write-str-multiple print)
  (flush))
```

File: README.md [hunk 4]
```diff
@@ -5,3 +5,30 @@ but contributions are welcome. It is targeted at folks who know Clojure better
...
+
+And used like this:
+
+    $ cat source.clj
+    (defn foo [x & xs] xs)
+
+    (inc 2)
+
+    (map inc [1 2 3])
+    $ cat source.clj | clj2el --multiple
```
Comment (borkdude) [inline README.md]: multiple flag can go?

Comment (teodorlu) [reply README.md]: Forgot that one, removing it. Thanks!

**Pull request merged**